mllib/src/main/scala/org/apache/spark/ml/Estimator.scala

```diff
@@ -82,5 +85,32 @@ abstract class Estimator[M <: Model[M]] extends PipelineStage {
     paramMaps.map(fit(dataset, _))
   }
 
+  @Since("2.3.0")
+  def fit(dataset: Dataset[_], paramMaps: Array[ParamMap],
```

Comment:
I will add doc for this interface later. But it can be reviewed first, currently this interface looks a little ugly.